internal/backend/svc/svc.go

```diff
@@ -285,11 +289,14 @@ func makeFxOpts(cfg *Config, opts RunOptions) []fx.Option {
 				http.Redirect(w, r, uiAddr, http.StatusFound)
 			})
 		}),
-		fx.Invoke(func(z *zap.Logger, lc fx.Lifecycle, muxes *muxes.Muxes) {
+		fx.Invoke(func(z *zap.Logger, lc fx.Lifecycle, muxes *muxes.Muxes, h healthreporter.HealthReporter) {
 			var ready atomic.Bool
 
 			muxes.NoAuth.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
-				// TODO(ENG-530): check db, temporal, etc.
+				if healthErrors := h.Report(); len(healthErrors) != 0 {
```

Comment:
this is not a health page, it's health endpoint that suppose to return 200 or 500 so deployments manager can decide if it should kill the instance or not

you are talking about status page that shows to users that status of every component in the system

Reply:
Good overview!

I did mean health and not status pages: Google's internal `/healthz` **always** serve (relevant metrics) as long as the process is alive, and the control plane (Borg) decides what to do with overloaded or misbehaving instances. Otherwise, how can you know **what** is wrong?

The only thing I'd clarify in your overview is that, for AK, readiness failures mean replacing the pod as soon as the failure is noticed by the K8s control plane.

So taking this comment back to this PR, does that mean that this PR doesn't make any claims about readiness? That's the part I'm confused about. We both see "health" as an advanced and complex aspect of "readiness", and I interpret that as "if not healthy then obviously also not ready, i.e. replace it", but it seems like this PR says instead "bad = not healthy or not ready, which are unrelated to each other".

Reply:
NM, I re-read your additions in Linear, and we had inverted terminology:

> This is not /ready implementation which mainly used to decide if service is ready to serve traffic (service could be healthy but under load in which case health would report ok but ready would report error)

I don't know for certain which one is correct, but either way it's not a blocker for this PR.

Reply:
this is an example from k8s docs (https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)
although they return the error in the response as a string, the main check if the http status code and the text is ignored by the deployment controller

```go
http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
    duration := time.Now().Sub(started)
    if duration.Seconds() > 10 {
        w.WriteHeader(500)
        w.Write([]byte(fmt.Sprintf("error: %v", duration.Seconds())))
    } else {
        w.WriteHeader(200)
        w.Write([]byte("ok"))
    }
})
```

you can see the definition for liveness (health) / readiness they have match what I know as standard